crates/oxc_linter/src/rules/eslint/no_console.rs

```diff
@@ -0,0 +1,120 @@
+use oxc_ast::{ast::Expression, AstKind};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-console): Unexpected console statement.")]
+#[diagnostic(severity(warning))]
+struct NoConsoleDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoConsole {
+    /// A list of methods allowed to be used.
+    ///
+    /// ```javascript
+    /// // allowed: ['info']
+    /// console.log('foo'); // will error
+    /// console.info('bar'); // will not error
+    /// ```
+    pub allow: Vec<String>,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallows using the global console object.
+    ///
+    /// ### Why is this bad?
+    /// In JavaScript that is designed to be executed in the browser,
+    /// it’s considered a best practice to avoid using methods on console.
+    /// Such messages are considered to be for debugging purposes and therefore
+    /// not suitable to ship to the client.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// console.log('here');
+    /// ```
+    NoConsole,
+    restriction
+);
+
+impl Rule for NoConsole {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        Self {
+            allow: value
+                .get(0)
+                .and_then(|v| v.get("allow"))
+                .and_then(serde_json::Value::as_array)
+                .map(|v| {
+                    v.iter()
+                        .filter_map(serde_json::Value::as_str)
+                        .map(ToString::to_string)
+                        .collect()
+                })
+                .unwrap_or_default(),
+        }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::ExpressionStatement(expr) = node.kind() {
+            if let Expression::CallExpression(call_expr) = &expr.expression {
```

Comment:
You can omit the first `AstKind::ExpressionStatement`. The visitor visits every single AST node, which means `node.kind()` can be anything.

It may be helpful to take a look at this example to understand what I mean. https://github.com/web-infra-dev/oxc/blob/main/crates/oxc_linter/examples/linter.rs#L40-L41

Reply:
By the way, I think a simple `(console.log())` will fail.